reader_concurrency_semaphore.cc

```diff
@@ -74,10 +74,20 @@ bool reader_concurrency_semaphore::try_evict_one_inactive_read() {
     return true;
 }
 
+future<lw_shared_ptr<reader_concurrency_semaphore::reader_permit>>
+reader_concurrency_semaphore::make_queue_overflow_exception_future() const {
+    return make_exception_future<lw_shared_ptr<reader_permit>>(
+            std::make_exception_ptr(std::runtime_error(
+                    format("{}: restricted mutation reader queue overload", _name))));
+}
+
 future<lw_shared_ptr<reader_concurrency_semaphore::reader_permit>> reader_concurrency_semaphore::wait_admission(size_t memory,
         db::timeout_clock::time_point timeout) {
     if (_wait_list.size() >= _max_queue_length) {
-        return make_exception_future<lw_shared_ptr<reader_permit>>(_make_queue_overloaded_exception());
+        if (_prethrow_action) {
+            _prethrow_action();
+        }
+        return make_queue_overflow_exception_future();
```

Comment:
nitpick: the content of this function might as well be inlined here.

Reply:
Done (inlined)